pkg/handler/iam_handler.go

```diff
@@ -219,3 +233,17 @@ func toBindingsMap(bs []*v1alpha1.Binding) map[string]map[string]struct{} {
 	}
 	return result
 }
+
+func expired(exp string) (bool, error) {
+	// An valid expiration expression will look like "request.time < timestamp('2023-06-14T19:55:03Z')".
+	matches := regexp.MustCompile(`request.time < timestamp\('([^']+)'\)`).FindStringSubmatch(exp)
```

Comment:
see [here](https://github.com/abcxyz/access-on-demand/blob/main/pkg/handler/iam_handler.go#L194)

Reply:
It's a little safer to move the regex compilation to a global, like this:

```
var expirationRegex = regex.MustCompile(...)

func expired(...) {
```

This way, the regex will compiled when the program starts. This has two benefits:
 - it's automatically tested. If the regex fails to compile, then it will panic, and any tests that link to this package will fail.
 - it's only compiled once, not every time the function is called (not so important in this case where a few microseconds don't matter)

Reply:
Make sense, updating!

Reply:
To keep you unblocked, I'm fine if you want to just submit this PR